src/indexing/EventIndex.ts

```diff
@@ -241,8 +241,10 @@ export default class EventIndex extends EventEmitter {
      * Listens for timeline resets that are caused by a limited timeline to
      * re-add checkpoints for rooms that need to be crawled again.
      */
-    private onTimelineReset = async (room: Room, timelineSet: EventTimelineSet, resetAllTimelines: boolean) => {
-        if (room === null) return;
+    private onTimelineReset = async (
+        room: Room | undefined | null,
```

Comment:
The signature in event-timeline-set.ts sayd `room: Room | undefined` - not nullable